test/runtest.py

```diff
@@ -2607,3 +2607,20 @@ def test_pandas_parquet_serialization():
     pd.DataFrame({"col1": [0, 1], "col2": [0, 1]}).to_parquet(filename)
     # Clean up
     shutil.rmtree(tempdir)
+
+
+def test_socket_directory_none_existant(shutdown_only):
+    level1_name = ray.ObjectID(_random_string()).hex()
+    level2_name = ray.ObjectID(_random_string()).hex()
+    temp_raylet_socket_dir = "/tmp/{}/{}".format(level1_name, level2_name)
```

Comment:
nit, let's put the socket under something like `/tmp/ray/tests` to avoid creating too many files in `/tmp`